### PR TITLE
refactor: Disclosure component

### PR DESCRIPTION
**Pull request by chienleng**
## Description

* remove top border
* rename title to headerTitle
* add more flexibility for header title styling
💁‍

## Related issues

https://app.shortcut.com/qantasaccommodation/story/78940/use-additional-information-accordion-for-current-content
💪

## Looks like this

Screen Shot 2021-10-05 at 10 14 28 am

🤨

---

**PR check-list**

🚨 **IMPORTANT:** Please review the [guidelines for contributing](./CONTRIBUTING.md) to this repository first! 👀

- [x] You've requested a review from at least ONE person. Preferably a roo-ui maintainer ([Philip](https://github.com/philipwindeyer), [Eric](https://github.com/eneo5541), [Mike](https://github.com/memcc), [Marcel](https://github.com/Anthropic), [Sam](https://github.com/SamuSan) [Long](https://github.com/KieraDOG)) or a Roo UI working-group human.
  - Note: If your PR has not been reviewed within two days of requesting a review, feel free to seek reviews from other [team members of Qantas Hotels](https://github.com/orgs/hooroo/people).
- [x] Your final merge commit message is in the format `<type>(<scope>): <description>` as per [the contributing guide](https://github.com/hooroo/roo-ui/blob/master/.github/CONTRIBUTING.md#commit-naming)
  - Example: `fix(dependencies): minor version bump to resolve security vulnerability in set-value`

### CODE REVIEW

**Comment by ehyland** [comment]
❌ Deploy Preview for *roo-ui* failed.


🔨 Explore the source changes: 97e5075817abf0eb677b359912cb49552138691a

🔍 Inspect the deploy log: [https://app.netlify.com/sites/roo-ui/deploys/615b8af1e40316000872e26e](https://app.netlify.com/sites/roo-ui/deploys/615b8af1e40316000872e26e?utm_source=github&utm_campaign=bot_dl)

File: src/components/Disclosure/Disclosure.js [hunk 1]
```diff
@@ -78,23 +91,27 @@ const Disclosure = (props) => {
...
+        {children}
       </Box>
     </StyledBox>
   );
 };
 
 Disclosure.defaultProps = {
   expand: true,
+  headerTitleSize: [rem('18px'), rem('22px')],
+  headerTitleWeight: 'normal',
```
Comment (KieraDOG) [inline src/components/Disclosure/Disclosure.js:103]: This 2 props looks bit `props drill`, how about pass in a `title` node?

File: src/components/Disclosure/Disclosure.story.js [hunk 2]
```diff
@@ -17,14 +17,14 @@ storiesOf('Components/Disclosure', module)
...
-        <Disclosure title="Heading">{children}</Disclosure>
+        <Disclosure headerTitle="Heading">{children}</Disclosure>
       </Box>
     </Container>
   ))
   .add('with panel hidden', () => (
     <Container>
       <Box p={3}>
-        <Disclosure id="my-id" title="Heading" expand={false}>
+        <Disclosure id="my-id" headerTitle="Heading" expand={false}>
```
Comment (KieraDOG) [inline src/components/Disclosure/Disclosure.story.js:27]:
```suggestion
        <Disclosure id="my-id" title={<Text fontWeight="bold>Heading</Text>} expand={false}>
```

**Review by eneo5541 — COMMENTED** [review]
I'm not sure how I feel about explicitly setting each header style. It feels like it could quickly get more bloated if we come across use cases where we need to add more styling to the header or panel. 

How do you feel about having the `panelAttrs` and `headerAttrs` as props and just spreading those into their respective components?
```
const Disclosure = ({ headerAttrs, panelAttrs, ...rest }) => {
    ...
    return (
        <StyledBox {...rest}>
            <StyledHeading {...headerAttrs} id={id ? `${id}-header` : undefined} />
            <Box {...panelAttrs} id={id ? `${id}-panel` : undefined}  />
        </StyledBox>
    );
};
```

File: src/components/Disclosure/Disclosure.story.js (hunk 2 above)
Comment (chienleng) [reply src/components/Disclosure/Disclosure.story.js:27]: `title` is rendered as a tooltip in browser, not sure that's the behaviour we want, also some concerns with accessibility (https://www.a11yproject.com/posts/2013-04-22-title-attributes/)

@eneo5541 's suggestion might be better, not sure I like another html tag just to change header styling.

**Comment by chienleng** [comment]
> I'm not sure how I feel about explicitly setting each header style. It feels like it could quickly get more bloated if we come across use cases where we need to add more styling to the header or panel.
> 
> How do you feel about having the `panelAttrs` and `headerAttrs` as props and just spreading those into their respective components?
> 
> ```
> const Disclosure = ({ headerAttrs, panelAttrs, ...rest }) => {
>     ...
>     return (
>         <StyledBox {...rest}>
>             <StyledHeading {...headerAttrs} id={id ? `${id}-header` : undefined} />
>             <Box {...panelAttrs} id={id ? `${id}-panel` : undefined}  />
>         </StyledBox>
>     );
> };
> ```

That works for me. Initial thoughts were that the header style would be the same, but it's not.

File: src/components/Disclosure/Disclosure.js [hunk 3]
```diff
@@ -30,15 +29,25 @@ const StyledIcon = styled(Icon)`
...
+  expand: initialExpand,
+  id,
+  onToggle,
+  headerTitle,
+  headerTitleSize,
+  headerTitleWeight,
+  borderColor,
+  children,
+  ...rest
+}) => {
```
Comment (philipwindeyer) [inline src/components/Disclosure/Disclosure.js:42]: Need to update the `DisclosureKnownProps` typedef to match this as well

**Review by philipwindeyer — APPROVED** [review]
Aside from the out-of-date props typedef, LGTM!

File: src/components/Disclosure/Disclosure.story.js (hunk 2 above)
Comment (KieraDOG) [reply src/components/Disclosure/Disclosure.story.js:27]: All good to me, just not a big fun to property drills

**Pull request merged**